haystack/pipeline.py

```diff
@@ -594,6 +721,130 @@ def run(self, **kwargs):
 
 
+class SklearnQueryClassifier(BaseComponent):
+    """
+    A node to choose between two nodes based on query classification result using Sklearn GradientBoosted models.
+
+    This node by default classifies between keyword and question/statement queries, read more about the dataset it was trained
+    on here:
+    https://ext-models-haystack.s3.eu-central-1.amazonaws.com/gradboost_query_classifier/readme.txt
+
+    If you want to classify between question queries and statement queries then use the following:
+
+    `query_classifier`:
+    https://ext-models-haystack.s3.eu-central-1.amazonaws.com/gradboost_query_classifier_statements/model.pickle
+
+    `query_vectorizer`
+    https://ext-models-haystack.s3.eu-central-1.amazonaws.com/gradboost_query_classifier_statements/vectorizer.pickle
+
+    Or else you can load another query classifier from local file.
+
+    Read more about the dataset it was trained on here:
+    https://ext-models-haystack.s3.eu-central-1.amazonaws.com/gradboost_query_classifier_statements/readme.txt.
+    """
+
+    outgoing_edges = 2
+
+    def __init__(
+        self,
+        query_classifier: Union[
+            str, Any
+        ] = "https://ext-models-haystack.s3.eu-central-1.amazonaws.com/gradboost_query_classifier/model.pickle",
+        query_vectorizer: Union[
+            str, Any
+        ] = "https://ext-models-haystack.s3.eu-central-1.amazonaws.com/gradboost_query_classifier/vectorizer.pickle",
+    ):
+        """
+        :param query_classifier: Gradient boosting based binary classifier to classify between keyword vs statement/question
+        queries or statement vs question queries.
+        :param query_vectorizer: A ngram based Tfidf vectorizer for extracting features from query.
+        """
+        if (
+            (not isinstance(query_classifier, Path))
+            and (not isinstance(query_classifier, str))
+        ) or (
+            (not isinstance(query_vectorizer, Path))
+            and (not isinstance(query_vectorizer, str))
+        ):
+            raise TypeError(
+                "query_classifier and query_classifier must either be of type Path or str"
+            )
+
+        if isinstance(query_classifier, Path):
+            file_url = urllib.request.pathname2url(r"{}".format(query_classifier))
+            query_classifier = f"file:{file_url}"
+
+        if isinstance(query_vectorizer, Path):
+            file_url = urllib.request.pathname2url(r"{}".format(query_vectorizer))
+            query_vectorizer = f"file:{file_url}"
+
+        self.query_classifier = pickle.load(urllib.request.urlopen(query_classifier))
+
+        self.query_tokenizer = pickle.load(urllib.request.urlopen(query_vectorizer))
+
+    def run(self, **kwargs):
+        query_vector = self.query_tokenizer.transform([kwargs["query"]])
+
+        is_question: bool = self.query_classifier.predict(query_vector)[0]
+        if is_question:
+            return (kwargs, "output_1")
+        else:
```

Comment:
Shall we change the output name here from "output_1" -> "semantic_query" and "output_2" -> "keywords"?
I think this will be help a lot when you stick pipelines together and can rather refer to `.keywords` as the input: 
```
    pipeline.add_node(name="elastic",component=elastic_retriever, inputs["SkQueryKeywordQuestionClassifier.keywords"])
```
Can you test quickly if this works already as expected or if anything in our pipeline class is currently blocking this? If yes, we can add two arguments to init (output_1_name, output_2_name) with the above defaults. This would allow users to switch it easily to their class names (e.g. when using the question vs statement model).

Reply:
@tholor the output follows a specific format where prefix always has to be "output_" and full format is something like "output_{integer}", so I can't change the output to that there, I have added those two in init, please let me know if that's the thing you wanted there.

Reply:
Ok my bad, sorry. I thought we already relaxed this requirement about the output format. 
Then let's stick with your original version (hard coded output_1 and output_2; no init args) and document in the docstring which output belongs to which category for the available models.

Reply:
@oryx1729 do you think we can (in general) relax the naming requirements for node outputs easily or do you see bigger drawbacks?